pkg/channel/consolidated/reconciler/controller/kafkachannel.go

```diff
@@ -568,10 +568,22 @@ func (r *Reconciler) deleteTopic(ctx context.Context, channel *v1beta1.KafkaChan
 
 func (r *Reconciler) updateKafkaConfig(ctx context.Context, configMap *corev1.ConfigMap) {
 	logger := logging.FromContext(ctx)
+
+	if r == nil {
+		// This typically happens during startup and can be ignored
+		return
+	}
+
+	if configMap == nil {
+		logger.Warn("Nil ConfigMap passed to configMapObserver; ignoring")
+		return
+	}
+
 	logger.Info("Reloading Kafka configuration")
 	kafkaConfig, err := utils.GetKafkaConfig(configMap.Data)
 	if err != nil {
 		logger.Errorw("Error reading Kafka configuration", zap.Error(err))
+		return
```

Comment:
Should you return an error here?

Reply:
I will give it a try

Reply:
This function is not called within the reconciliation loop. It is called when config-kafka is changed.
A global resync is done later.

Reply:
ah ok gotcha!